OpenRA.Mods.Common/MapGenerator/MapGeneratorSettings.cs

```diff
@@ -13,389 +13,237 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Support;
 
 namespace OpenRA.Mods.Common.MapGenerator
 {
-	public sealed class MapGeneratorSettings
+	public abstract class MapGeneratorOption
 	{
-		/// <summary>How an option should be treated for UI purposes.</summary>
-		public enum UiType
+		[FieldLoader.Ignore]
+		public readonly string Id;
+		public readonly string Label = null;
+		public readonly int Priority = 0;
+
+		protected MapGeneratorOption(string id, MiniYaml yaml)
 		{
-			Hidden,
-			DropDown,
-			Checkbox,
-			Integer,
-			Float,
-			String,
+			Id = id;
+			FieldLoader.Load(this, yaml);
 		}
 
-		/// <summary>Represents a bunch of settings, along with UI information.</summary>
-		public sealed class Choice
+		public abstract IReadOnlyCollection<MiniYamlNode> GetSettings(ITerrainInfo terrainInfo);
+	}
+
+	public class MapGeneratorBooleanOption : MapGeneratorOption
+	{
+		[FieldLoader.Require]
+		public readonly string Parameter = null;
```

Comment:
Is `Parameter` used? Is it not redundant with the `@` suffix in the map-generators.yaml?

Reply:
This sets the parameter that is modified by the checkbox. Its the equivalent to the multi-choice `Settings`, which is distinct from the @suffix.